### PR TITLE
Adding status field for backuplocation.

### PR DESCRIPTION
**Pull request by siva-portworx**
Adding status field for backuplocation to store Deleting or Delete Pending status.

### CODE REVIEW

File: pkg/apis/v1/api.proto [hunk 1]
```diff
@@ -258,7 +258,21 @@ message BackupLocationInfo {
         (gogoproto.moretags) = "secure:\"true\""
     ];
     string cloud_credential = 4;
+    StatusInfo status = 5;
+    bool cloud_delete = 6;
```
Comment (prashanthpx) [inline pkg/apis/v1/api.proto]: Should you call these as `backup_delete`? If a user is hosting something like minio, the term cloud_delete could be confusing.

Comment (prashanthpx) [reply pkg/apis/v1/api.proto]: Or even like `delete_backups`

Comment (siva-portworx) [reply pkg/apis/v1/api.proto]: Made it as delete_backups.

File: pkg/apis/v1/api.proto [hunk 2]
```diff
@@ -258,7 +258,21 @@ message BackupLocationInfo {
...
+    StatusInfo status = 5;
+    bool delete_backups = 6;
     oneof config { S3Config s3_config = 100; }
+    // Message for maintaing status of the object.
+    message StatusInfo {
+        enum Status {
+            Invalid = 0;
+            Valid = 1;
+            Deleting = 2;
+            DeletePending = 3;
```
Comment (disrani-px) [inline pkg/apis/v1/api.proto]:
```suggestion
            Deleting = 2;
            DeletePending = 3;
```

Not clear why we need both. Add comments or use just one status

Comment (siva-portworx) [reply pkg/apis/v1/api.proto]: As discussed, keeping only DeletePending state alone

**Pull request merged**